seatunnel-connectors-v2/connector-redis/src/main/java/org/apache/seatunnel/connectors/seatunnel/redis/source/RedisSource.java

```diff
@@ -40,21 +38,23 @@
 import org.apache.seatunnel.connectors.seatunnel.redis.exception.RedisConnectorException;
 import org.apache.seatunnel.format.json.JsonDeserializationSchema;
 
-import com.google.auto.service.AutoService;
+import com.google.common.collect.Lists;
+
+import java.util.List;
 
-@AutoService(SeaTunnelSource.class)
 public class RedisSource extends AbstractSingleSplitSource<SeaTunnelRow> {
     private final RedisParameters redisParameters = new RedisParameters();
     private SeaTunnelRowType seaTunnelRowType;
     private DeserializationSchema<SeaTunnelRow> deserializationSchema;
 
+    private CatalogTable catalogTable;
+
     @Override
     public String getPluginName() {
-        return "Redis";
+        return RedisConfig.CONNECTOR_IDENTITY;
     }
 
-    @Override
-    public void prepare(Config pluginConfig) throws PrepareFailException {
+    public RedisSource(Config pluginConfig) {
```

Comment:
Please use `ReadOnlyConfig`, do not convert `ReadOnlyConfig` to `Config`. This is a bad example, we want connectors to be able to use `ReadOnlyConfig` instead of `Config`